include/datastructure/GeneralTimetable.hpp

```diff
@@ -0,0 +1,790 @@
+    return *this > other || *this == other;
+  }
+  bool operator!=(const GeneralScheduledStop& other) const {
+    return !(*this == other);
+  }
+
+  [[nodiscard]] bool conflicts(const GeneralScheduledStop& other) const {
+    // Same station name is also a conflict
+    if (station == other.station) {
+      return true;
+    }
+
+    // If there is a time where both have to stop, there is a conflict
+    const auto& interval1 = get_forced_stopping_interval();
+    const auto& interval2 = other.get_forced_stopping_interval();
+    if (interval1.first > interval1.second ||
+        interval2.first > interval2.second) {
+      return false;
+    }
+    return interval1.first <= interval2.second &&
+           interval2.first <= interval1.second;
+  }
+
+  [[nodiscard]] std::pair<int, int> get_forced_stopping_interval() const {
+    std::pair<int, int> interval = {begin.second, end.first};
+    if (begin.first + min_stopping_time > interval.second) {
+      interval.second = begin.first + min_stopping_time;
+    }
+    if (end.second - min_stopping_time < interval.first) {
+      interval.first = end.second - min_stopping_time;
+    }
+    return interval;
+  }
+
+  [[nodiscard]] const std::pair<int, int>& get_begin_range() const {
+    return begin;
+  }
+  [[nodiscard]] const std::pair<int, int>& get_end_range() const { return end; }
+  [[nodiscard]] int get_min_stopping_time() const { return min_stopping_time; }
+  [[nodiscard]] const std::string& get_station_name() const { return station; }
+
+  // Constructor
+  GeneralScheduledStop(std::pair<int, int> begin, std::pair<int, int> end,
+                       int min_stopping_time, std::string station)
+      : begin(std::move(begin)), end(std::move(end)),
+        min_stopping_time(min_stopping_time), station(std::move(station)) {
+    if (this->begin.second < this->begin.first) {
+      throw exceptions::InvalidInputException(
+          "Interval begin has negative length");
+    }
+    if (this->end.second < this->end.first) {
+      throw exceptions::InvalidInputException(
+          "Interval end has negative length");
+    }
+    if (this->min_stopping_time <= 0) {
+      throw exceptions::InvalidInputException(
+          "Minimum stopping time is non-positive");
+    }
+    if (this->begin.first < 0) {
+      throw exceptions::InvalidInputException(
+          "Interval begin has negative start time");
+    }
+    if (this->end.first < 0) {
+      throw exceptions::InvalidInputException(
+          "Interval end has negative start time");
+    }
+    if (this->end.second < this->begin.first) {
+      throw exceptions::InvalidInputException(
+          "Interval end starts before interval begin");
+    }
+    if (this->end.second - this->begin.first < this->min_stopping_time) {
+      throw exceptions::InvalidInputException(
+          "Maximal Interval is shorter than minimum stopping time");
+    }
+  }
+};
+
+class BaseGeneralSchedule {
```

Comment:
Just a question for general understanding. Is this really necessary since the class doesn't do anything? The compilation will fail anyway if the template used below doesn't expose the correct methods. Or is there some additional benefit to it?

Reply:
It is used because checks like static_assert(std::is_base_of_v<BaseGeneralSchedule, T>) did not seem to work as desired without this class.